mrt_file_server/views.py

```diff
@@ -100,11 +100,21 @@ def download_schematic_post():
     log_warn('SCHEMATIC_DOWNLOAD_FILENAME_WHITESPACE', filename)
     return
 
-  filename = "{}.schematic".format(secure_filename(filename))
-
-  if file_exists_in_dir(downloads_dir, filename):
-    log_info('SCHEMATIC_DOWNLOAD_SUCCESS', filename)
-    return send_from_directory(downloads_dir, filename, as_attachment = True)
+  filename1 = "{}.schematic".format(secure_filename(filename))
+  filename2 = "{}.schem".format(secure_filename(filename))
+
+  file1_exists = file_exists_in_dir(downloads_dir, filename1)
+  file2_exists = file_exists_in_dir(downloads_dir, filename2)
+
+  if file1_exists and not file2_exists:
+    log_info('SCHEMATIC_DOWNLOAD_SUCCESS', filename1)
+    return send_from_directory(downloads_dir, filename1, as_attachment = True)
+  elif not file1_exists and file2_exists:
+    log_info('SCHEMATIC_DOWNLOAD_SUCCESS', filename2)
+    return send_from_directory(downloads_dir, filename2, as_attachment = True)
+  elif file1_exists and file2_exists:
+    flash_by_key(app, 'SCHEMATIC_DOWNLOAD_FILENAME_NOT_UNIQUE', filename)
+    log_warn('SCHEMATIC_DOWNLOAD_FILENAME_NOT_UNIQUE', filename)
```

Comment:
Overall, I'm not a fan of this approach with checking for both ".schem" and ".schematic" files and downloading if only one exists. The download page also still says ".schematic" after the text field, so it is also not obvious to the use that they can download ".schem" files from the page.

I was thinking of doing one of the following approaches instead:
1. Remove the ".schematic" label from the page and make the user enter the full file name including ".schematic" or ".schem" suffix in the text field. This will require adding error checks for invalid file suffixes.
2. Replace the ".schematic" label with a combo box that can be switched from the two options ".schematic" or ".schem". This allows stricter control on what suffixes are allowed, but requires some code to be written for the combobox logic.

Either approach requires a bit of work, so if you'd like to try to tackle it, feel free to do so. Otherwise, I'll probably implement it myself in the near future.